gnovm/cmd/gno/test.go

```diff
@@ -500,10 +512,25 @@ func runTestFiles(
 	n := gno.MustParseFile("main_test.gno", testmain)
 	m.RunFiles(n)
 
+	printedEvents := 0
+
 	for _, test := range testFuncs.Tests {
 		testFuncStr := fmt.Sprintf("%q", test.Name)
 
-		eval := m.Eval(gno.Call("runtest", testFuncStr))
+		res := gno.Call("runtest", testFuncStr)
+
+		eval := m.Eval(res) // NOTE: verbose prints get here
+		if printEvents {
+			ctx := m.Context.(stdlibs.ExecContext)
+
+			events := ctx.EventLogger.Events()
+			for _, ev := range events[printedEvents:] {
+				printedEvents++
+				// XXX: print events with better formatting (e.g. JSON)
+				strEv := fmt.Sprint(ev)
+				io.ErrPrintfln("---       event: %s", colors.ColoredBytesOnlyAscii([]byte(strEv), colors.Magenta))
```

Comment:
Why is this an error print?

Reply:
99% of printing in that file uses error.

For example, metrics are printed using error
https://github.com/gnolang/gno/blob/9eeb61156d79208cd63c23bbe8c3a03806240f2b/gnovm/cmd/gno/test.go#L569